lib/context.js

```diff
@@ -17,7 +17,10 @@ function Context (executor, job, result, device) {
 	self.bin_dir = path.join(__rootdir, 'bin');
 	self.output_dir = result.output_dir;
 	self.working_dir = job.getWorkingDir();
-	if (device) self.device_identifier = device.identifier;
+	if (device) {
+		self.device_identifier = device.identifier;
+		self.device = device;
```

Comment:
Why hold `self.device_identifier` when you could just use `self.device.identifier`?